katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+        # All antennas in configuration as katpoint Antenna objects
+
+        ants = []
+        for antenna in config_group["Antennas"]:
+            name = config_group["Antennas"][antenna].attrs['name']
+            latitude = config_group["Antennas"][antenna].attrs['latitude']
+            longitude = config_group["Antennas"][antenna].attrs['longitude']
+            altitude = config_group["Antennas"][antenna].attrs['altitude']
+            diameter = config_group["Antennas"][antenna].attrs['diameter']
+            delay_model = None
+            pointing_model = make_pmodel_string(config_group["Antennas"][antenna]['pointing-model-params'])
+            beamwidth = config_group["Antennas"][antenna].attrs['beamwidth']
+
+            ants.append(katpoint.Antenna(name, latitude, longitude, altitude, diameter, delay_model, pointing_model, beamwidth))
+
+        # Temporarily casting this to string. I edited a file by hand to change the name and it seems to now read the
+        # attr as a np.ndarray
+        #ants = [katpoint.Antenna(str(config_group['Antennas'][name].attrs['description']))
+        #        for name in config_group['Antennas']]
+
+        self.subarrays = [Subarray(ants, corrprods)]
+        self.sensor['Observation/subarray'] = CategoricalData(self.subarrays, [0, len(data_timestamps)])
+        self.sensor['Observation/subarray_index'] = CategoricalData([0], [0, len(data_timestamps)])
+
+        # Store antenna objects in sensor cache too, for use in virtual sensor calculations
+        for ant in ants:
+            self.sensor['Antennas/%s/antenna' % (ant.name,)] = CategoricalData([ant], [0, len(data_timestamps)])
+
+        # ------ Extract spectral windows / frequencies ------
+        bandwidth = get_single_value(config_group['DBE'], 'bandwidth')
+        num_chans = get_single_value(config_group['DBE'], 'n_chans')
+        channel_width = bandwidth / num_chans
+
+        # Sky centre frequency = LO1 + LO2 - IF
+        band_select = self.sensor["RFE/rfe.band.select.LCP"]
+        LO_5GHz = self.sensor["RFE/rfe.lo-intermediate.5GHz.frequency"]
+        LO_6p7GHz = self.sensor["RFE/rfe.lo-intermediate.6_7GHz.frequency"]
+        LO_final = self.sensor["RFE/rfe.lo-final.frequency"]
+
+        centre_freq_5GHz = LO_5GHz + LO_final - 600e6;
+        centre_freq_6p7GHz = LO_6p7GHz + LO_final - 600e6;
+
+        centre_freq = np.zeros(len(band_select))
+        for i in range(len(centre_freq)):
+            if band_select[i] == "0":
+                centre_freq[i] = centre_freq_5GHz[i]
+            elif band_select[i] == "1":
+                centre_freq[i] = centre_freq_6p7GHz[i]
+            else:
+                raise BrokenFile("Unknown band selection, seems to be neither 5 GHz nor 6.7 GHz.")
+
+        # If we aren't in wideband mode, then we need to do some additional tweaking:
+        if (fine_size != 0):
+            centre_freq -= 200e6  # Because we're going to count from the bottom of the band.
```

Comment:
It's safer to use fractions of `sampling_frequency` here.

Reply:
Noted. Changed in 56e9a20.